dot_graph.go

```diff
@@ -0,0 +1,391 @@
+package seth
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/awalterschulze/gographviz"
+)
+
+func findShortestPath(calls []*DecodedCall) []string {
+	callMap := make(map[string]*DecodedCall)
+	for _, call := range calls {
+		callMap[call.CommonData.Signature] = call
+	}
+
+	var root *DecodedCall
+	for _, call := range calls {
+		if call.CommonData.ParentSignature == "" {
+			root = call
+			break
+		}
+	}
+
+	if root == nil {
+		return nil // No root found
+	}
+
+	var end *DecodedCall
+	for i := len(calls) - 1; i >= 0; i-- {
+		if calls[i].CommonData.Error != "" {
+			end = calls[i]
+			break
+		}
+	}
+	if end == nil {
+		end = calls[len(calls)-1]
+	}
+
+	type node struct {
+		call *DecodedCall
+		path []string
+	}
+
+	queue := []node{{call: root, path: []string{root.CommonData.Signature}}}
+	visited := make(map[string]bool)
+
+	for len(queue) > 0 {
+		currentNode := queue[0]
+		queue = queue[1:]
+
+		currentCall := currentNode.call
+		currentPath := currentNode.path
+
+		if currentCall.CommonData.Signature == end.CommonData.Signature {
+			return currentPath
+		}
+
+		visited[currentCall.CommonData.Signature] = true
+
+		for _, call := range calls {
+			if call.CommonData.ParentSignature == currentCall.CommonData.Signature && !visited[call.CommonData.Signature] {
+				newPath := append([]string{}, currentPath...)
+				newPath = append(newPath, call.CommonData.Signature)
+				queue = append(queue, node{call: call, path: newPath})
+			}
+		}
+	}
+
+	return nil // No path found
+}
+
+var defaultTruncateTo = 20
```

Comment:
Is that the maximum height of the tree, should it be a parameter?

Reply:
no, that's for truncating input values of functions as these sometimes are `[]uint` of varying length, but they easily can have > 100 elements and then the graph becomes really ugly and harder to read